### DISCUSSION TITLE
How to setup codeium

### DISCUSSION THREAD

**Discussion by AntCursor** [nix-community/nixvim#1216] [Q&A] [answered]
Hello 🙋, i want to use codeium with nixvim, first i enabled it by plugins.codeium-nvim.enable , then i used Codeium Auth command and authenticated correctly, but it wont do nothing, on any file type, how i can make it work?

**Comment by traxys** [comment, marked as answer]
You may need to add codeium as a cmp source as mentionned by the [README](https://github.com/Exafunction/codeium.nvim?tab=readme-ov-file#nvim-cmp)

This can be done with the following:
```
    plugins.cmp = {
      enable = true;

      settings.sources = [
        {name = "codeium";}
      ];
    };
```

More generally I'd suggest sticking to the upstream documentation, because maintainers don't know much on how each plugin really works, we mostly care about mapping lua to nix
